release-process: Add signing of tarball plus minor improvements

real-or-random opened the pull request.

real-or-random committed:
release-process: Refactor

real-or-random committed:
release-process: Add signing of tarball

Resolves #1175.

real-or-random committed:
release-process: Fix nits

real-or-random committed:
release-process: Add "cleaning up"

fanquake commented:
You might be interested in this downstream discussion: https://github.com/gentoo/gentoo/pull/43104#discussion_r2224178492.

fanquake commented:
Push the milestone here (and on #1175)?

theStack commented:
Concept ACK

> We should look into the https://github.com/bitcoin-core/bitcoin-maintainer-tools/blob/main/make-tag.py script. Not sure if it's immediately compatible with libsecp256k1 as it performs a few checks probably specific to Core but the idea to include a Treehash512 in the tag description is clever.

Took a stab at that, adapting the script to the versioning scheme that libsecp256k1 uses (i.e. fixed major.minor.patch without rcs): https://github.com/theStack/secp256k1/commits/make-tag-py_libsecp_edition/tools/make-tag.py
A quick test based on an extra commit that sets a fake version 0.23.42 and switches IS_RELEASE to true was successful:
<details>
<summary>make-tag.py libsecp edition experiments</summary>

```
$ ./tools/make-tag.py v0.23.42
libsecp256k1 0.23.42

Tree-SHA512: 4767cfb6d17034f166f3c3ab261c7434dbe610ffd9e0117cdc797f5bd24e2fb9dbb6a55a571c6874a23800589014de8c1d928e7c272309fa65be4985c5602abe

$ git show v0.23.42
tag v0.23.42
Tagger: Sebastian Falbesoner <sebastian.falbesoner@gmail.com>
Date:   Tue Jul 7 22:14:14 2026 +0200

libsecp256k1 0.23.42

Tree-SHA512: 4767cfb6d17034f166f3c3ab261c7434dbe610ffd9e0117cdc797f5bd24e2fb9dbb6a55a571c6874a23800589014de8c1d928e7c272309fa65be4985c5602abe
-----BEGIN PGP SIGNATURE-----
...
-----END PGP SIGNATURE-----

$ ./tools/treehash512.py 
4767cfb6d17034f166f3c3ab261c7434dbe610ffd9e0117cdc797f5bd24e2fb9dbb6a55a571c6874a23800589014de8c1d928e7c272309fa65be4985c5602abe
```
</details>

theStack commented:
ACK e6aa166c081b55b1cab71de092fb43b8a93c83d8

Proposal to address [@hebasto's suggestion](https://github.com/bitcoin-core/secp256k1/pull/1709#issuecomment-3319677866) to include the `check-abi` output in the PR description (in a minimum-diff way, could still be improved later):

```diff
diff --git a/doc/release-process.md b/doc/release-process.md
index c1ddef1a..0e43e61c 100644
--- a/doc/release-process.md
+++ b/doc/release-process.md
@@ -57,7 +57,7 @@ You can see your default key by running `echo "test" | gpg --sign --verbose > /d
    * if this is not a patch release,
        * updates `_PKG_VERSION_*` and `_LIB_VERSION_*`  in `configure.ac`, and
        * updates `project(libsecp256k1 VERSION ...)` and `${PROJECT_NAME}_LIB_VERSION_*` in `CMakeLists.txt`.
-2. Perform the [sanity checks](#sanity-checks) on the PR branch.
+2. Perform the [sanity checks](#sanity-checks) on the PR branch, and add the output of the [`check-abi`](/tools/check-abi.sh) tool (screenshot of the generated compatibility HTML report) to the PR description.
 3. After the PR has been merged, tag the commit, and push the tag:
    ```
    RELEASE_COMMIT=<merge commit of step 1>
```

real-or-random committed:
release-process: Add attaching output of check-abi.sh to PR

real-or-random commented:
Thanks a lot. Yes, a screenshot seems very pragmatic. Updated in a similar way.

> adapting the script to the versioning scheme that libsecp256k1 uses (i.e. fixed major.minor.patch without rcs): [theStack/secp256k1@`make-tag-py_libsecp_edition`/tools/make-tag.py (commits)](https://github.com/theStack/secp256k1/commits/make-tag-py_libsecp_edition/tools/make-tag.py)
> A quick test based on an extra commit that sets a fake version 0.23.42 and switches IS_RELEASE to true was successful:

I think it's a bit late now for the upcoming release (we'll need to figure out whether we want to add the script to the repo, etc.) But this sounds like a good improvement for the release after that.

theStack reviewed (approved):
re-ACK 23859af39fe03640cd9e774133d55c562f59cc6c

hebasto reviewed (approved):
ACK 12d9cfd86e77629b5e7219c6c5407ab8a5221209.

theStack reviewed (approved):
re-ACK 12d9cfd86e77629b5e7219c6c5407ab8a5221209
